src/components/DiagramFrame/SeqDiagram/LifeLineLayer/Participant.vue

```diff
@@ -36,17 +36,21 @@ import iconPath from '../../Tutorial/Icons';
 import useIframePositionInParent from '@/functions/useIframePositionInParent';
 import { computed, ref } from 'vue';
 import useDocumentScroll from '@/functions/useDocumentScroll';
+import { useStore } from 'vuex';
 
 export default {
   name: 'Participant',
   setup(props) {
+    const store = useStore()
     const participant = ref(null)
     const [intersectionTop] = useIframePositionInParent()
     const [scrollTop] = useDocumentScroll()
     const translate = computed(() => {
-      const top = intersectionTop.value + scrollTop.value
+      let top = intersectionTop.value + scrollTop.value
+      if (intersectionTop.value > 1 && store.state.stickyOffset) top += store.state.stickyOffset
       if (top < props.offsetTop + 50) return 0
-      return top - props.offsetTop - 50
+      const diagramHeight = document.querySelector('pre.zenuml')?.clientHeight || 0
```

Comment:
We cannot assume the root is `pre.zenuml`. We need a different way to get diagramHight.